### PR TITLE
Support route compilation (fixes #14)

### LINKED ISSUE #14: compile routes for snappy real-life use
When a `Match` is found, it should contain directly a optionally compiled handler to be invoked by the client.

### PR DESCRIPTION
**Pull request by ikitommi**
* also, so docs

### CODE REVIEW

File: src/reitit/core.cljc [hunk 1]
```diff
@@ -55,18 +55,25 @@
...
 
-(defn resolve-routes [data {:keys [coerce] :or {coerce identity} :as opts}]
-  (->> (walk data opts)
-       (map-meta merge-meta)
-       (mapv (partial coerce))
-       (filterv identity)))
+(defn resolve-routes [data {:keys [coerce] :as opts}]
+  (cond-> (->> (walk data opts)
+               (map-meta merge-meta))
+          coerce (->> (mapv (partial coerce))
```
Comment (miikka) [inline src/reitit/core.cljc]: Could be simpler and maybe even more performant with `(cond->> ... coerce (into [] (keep (partial coerce))))`.

File: test/cljc/reitit/core_test.cljc [hunk 2]
```diff
@@ -55,25 +55,44 @@
...
+                 (reitit/routes router))))
+        (testing "route match contains compiled handler"
+          (is (= 2 @compile-times))
+          (let [{:keys [handler]} (reitit/match-by-path router "/api/pong")]
+            (is handler)
+            (is (= "/api/pong" (handler)))
+            (is (= 2 @compile-times))))))
+    (testing "default compile"
+      (let [router (reitit/router ["/ping" (constantly "ok")])]
+        (println (reitit/match-by-path router "/ping"))
```
Comment (miikka) [inline test/cljc/reitit/core_test.cljc]: [nit] debug print

Comment (ikitommi) [inline test/cljc/reitit/core_test.cljc]: removed in ring-branch (on top of this)

File: src/reitit/core.cljc (hunk 1 above)
Comment (ikitommi) [inline src/reitit/core.cljc]: 👍

**Pull request merged**